keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1206,6 +1206,33 @@ class SqlArtifactRepository(
     }
   }
 
+  override fun getVersionInfoInEnvironment(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String,
+    artifact: DeliveryArtifact
+  ): List<StatusInfoForArtifactInEnvironment> =
+    sqlRetry.withRetry(READ) {
+      jooq
+        .select(
```

Comment:
My instinct is to add a limit to this query. I can imagine this returning a very large number of records.